kernel/interrupt.rs

```diff
@@ -39,8 +39,9 @@ pub fn handle_irq(irq: u8) {
     let handler = &mut IRQ_HANDLERS.lock()[irq as usize];
     unsafe {
         (*handler.assume_init_mut())();
-        // FIXME: Temporarily moved to here to avoid dead locking in receive_ethernet_frame.
-        //        To be replaced soon with a new mechanism like softirq.
-        process_packets();
     }
+
+    // So-called "bottom half" in Linux kernel. Execute time-consuming but
+    // non-critical work like processing packets.
+    run_deferred_jobs();
```

Comment:
I do not understand how this is different than processing packets in the handler? 
IRQ_HANDLERS are still locked, right?

Just the idea, maybe DeferredJob can be processed in the idle thread?

Reply:
Good point! I missed that. Will fix.

> Just the idea, maybe DeferredJob can be processed in the idle thread?

I think so too. We should run `run_deferred_jobs` from the idle loop if we limit # of jobs run in the interrupt handler.